intake/catalog/base.py

```diff
@@ -717,6 +730,14 @@ def _load(self):
                  for source in info['sources']})
 
     def search(self, *args, **kwargs):
+        # if callable do special callable thing
+        if hasattr(args[0], '__call__'):
+            # check function against all of them
+            # TODO: run proper walk when walk doesn't take ages
+            entries = {k: self[k] for k, v in self._entries.items() if args[0](v.describe())}
```

Comment:
This doesn't do walk like the local one. I left this out because `walk` was rather slow. I think this is because it is hitting the catalog refresh each time, maybe?

Reply:
I think that's OK: your function could filter whole catalogs in/out too.